src/validation.cpp

```diff
@@ -688,10 +686,7 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         return false; // state filled in by CheckTxInputs
     }
 
-    // If fee_out is passed, return the fee to the caller
-    if (args.m_fee_out) {
-        *args.m_fee_out = nFees;
-    }
+    ws.m_base_fees = nFees;
```

Comment:
Do we even need the local `nFees` variable, now that `ws.m_base_fees` is always set? Can you just pass a reference to `ws.m_base_fees` to `CheckInputs()`?

Reply:
True, done